tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -39,6 +39,42 @@ def sync(sync_ttl):
             sds_details = plugin.discover_storage_system()
             if sds_details is None:
                 break
+
+            if "peers" in sds_details and NS.tendrl_context.integration_id:
+                _cluster = NS.tendrl.objects.Cluster(
+                    integration_id=NS.tendrl_context.integration_id
+                ).load()
+                this_peer_uuid = ""
+                if _cluster.is_managed != "yes":
+                    for peer_uuid, data in sds_details.get("peers", {}):
+                        peer = NS.tendrl.objects.GlusterPeer(
+                            peer_uuid=peer_uuid,
+                            hostname=data['hostname'],
+                            connected=data['connected']
+                        )
+                        peer.save()
+                        if data['hostname'] == "localhost":
+                            this_peer_uuid = peer_uuid
+
+                    # Figure out the hostname used to probe this peer
+                    integration_id_index_key = \
+                        "indexes/tags/tendrl/integration/%s" %\
+                        NS.tendrl_context.integration_id
+                    _node_ids = etcd_utils.read(integration_id_index_key).value
+                    _node_ids = json.loads(_node_ids)
+                    peer_node_id = ""
+                    for _node_id in _node_ids:
+                        if _node_id != NS.node_context.node_id:
+                            peer_node_id = _node_id
+                            break
+
+                    if peer_node_id:
```

Comment:
Is it like in one peer we maintain one other's peer's uuid? Not getting clear to me what purpose it serves. Can you please add some code level comments and explain?

Reply:
Every node agent will save the gluster pool list. And to identify the hostname for current node we query any other node in cluster to find out

Reply:
Where is this pool list saved?

Reply:
Check this PR and this snippet

https://github.com/Tendrl/commons/pull/891
https://github.com/Tendrl/commons/pull/891/files#diff-1a5320aa0fd1eea178e6ed255509c9c4R1192

Reply:
Ah yes. Get that point.

Reply:
So all nodes have gluster peer object and to avoid localhost name we are reading gluster peer object for the current node from another node. This is my understanding.

Reply:
looks good to me